tests/utils/test_password_manager.py

```diff
@@ -330,3 +330,32 @@ def test_get_pypi_token_with_env_var_not_available(
     result_token = manager.get_pypi_token(repo_name)
 
     assert result_token is None
+
+
+def test_disabled_keyring_never_called(
+    config: Config, with_simple_keyring: None, dummy_keyring: DummyBackend
+) -> None:
+    config.config["keyring"]["enabled"] = False
+    config.config["http-basic"] = {"onlyuser": {"username": "user"}}
+
+    manager = PasswordManager(config)
+    num_public_functions = len([f for f in dir(manager) if not f.startswith("_")])
+    if num_public_functions != 10:
+        pytest.fail(
+            f"A function was added to or removed from the {PasswordManager.__name__} "
+            "class without reflecting this change in this test."
+        )
+
```

Comment:
Maybe, we can add the following line in this test, too:

```
with pytest.raises(PoetryKeyringError):
        _ = manager.keyring
```

and then add a comment like "No method must try to access keyring. If it does a PoetryKeyringError will be raised." to make clear why there is no assert in the following lines.

Reply:
If I add this to this test, I would remove the other test, because config test just becomes a subset of this test.
What about a comment that states that the test in test_config makes sure that access `.keyring` raises an error and we just check that accessing the important functions of the password manager don't raise an exception.

Reply:
I would probably even do it the other way around and check for the exception only in `test_password_manager`. The goal of `test_config` is only to check that the config option is propagated correctly. It's sufficient that `use_keyring` is `False` before and `True` after setting the option. All implications of `use_keyring` being `False` should be checked in `test_password_manager`.

Apart from that, I think even if there is some duplication, we should keep both tests because it corresponds to our (unwritten) test concept. If we introduce a new config option:

- We want a test in `test_config` that just checks that the option is propagated correctly. Other effects of the option do not matter here.
- We want a test of the changed behavior in the unit test of the influenced module. In this test it does not matter how the behavior is triggered (via config option or directly if there is an attribute that can be set). Just setting `password_manager.use_keyring = False` would be fine if that's sufficient. (Not saying you should do this, it just does not matter.)

Reply:
Ok, I changed it